### PR TITLE
[1.10] Use cmake for building libcurl

### PR DESCRIPTION
**Pull request by ligurio**

### CODE REVIEW

File: CMakeLists.txt
Comment (Totktonada) [inline CMakeLists.txt:314]: It looks as a reconstruction of 4bd44330157819c85ece060c1186cff41bae1ee1 logic, but a bit differently. I would rather backport the original commit: it provides our own FindZLIB.cmake (so it will not be changed by a cmake update -- AFAIK we do that for all Find\*.cmake files) and uses the same paths everywhere across the build scripts.

In such cases I rewrite the original commit message with an explanation why it is backported, see d8460466680f07f7cd75c4a1b273f20a18a2f3a5 for example.

Please, place the fix before the BuildLibCURL.cmake update: it is desirable to keep every commit working to make `git bisect` easier.

File: cmake/BuildLibCURL.cmake
Comment (Totktonada) [inline cmake/BuildLibCURL.cmake:3]: There were a couple of fixes in `master`: at least 9d620bd03c7a7f4b93e982f5eccf12afa481a4de and ed16c1e5780cf355ab9bf923ee6f9daf5b6363f8. We should reflect them before the merge.

I also see libnghttp2 linking code below, but here we have no the nghttp2 submodule in 1.10. So I compared all commits in 1.10 and master, which touch this file:

| Commit ID (master) | Description                                         | Commit ID (1.10)                                |
| ------------------ | --------------------------------------------------- | ----------------------------------------------- |
| ed16c1e57          | build: fix bundled libcurl and c-ares build on OS X | Not backported                                  |
| 9d620bd03          | http client: return back CA certificates search     | Not backported                                  |
| 5226027e7          | third_party: update libcurl from 7.83.0 to 7.84.0   | d0fe7f449 + the option in this commit               |
| c92a1699d          | cmake: support build using Ninja                    | In this commit, only for libcurl                |
| 1810ddfde          | third_party: update libcurl from 7.80.0 to 7.83.0   | ffe6d113a                                       |
| fa8d70caa          | build: link bundled libcurl with nghttp2            | In this commit, but without the submodule       |
| 4bd443301          | cmake: factor out FindZLIB                          | In this PR, but differently (see comment above) |
| 6ab951fd5          | build: fix linking of debug curl library            | In this commit                                  |
| 01964d4cd          | build: build libcurl with PIC enabled               | In this commit                                  |
| 4bde1dbc0          | build: enable smtp                                  | d53ba05c4                                       |
| 820d2be66          | build: fix configuring using cmake3 command         | In this commit                                  |
| 2b0760192          | build: enable cmake in curl build                   | In this commit                                  |
| 800e5ed61          | build: refactor static build process                | BuildLibCURL.cmake are reflected this commit, but the rest is not backported yet deliberately |
| 238370761          | build: link bundled libcurl with c-ares             | 9987838ab                                       |
| 0bead6004          | build: don't pass LDFLAGS from environment to curl  | b5744f34b                                       |
| 1eead75e2          | build: pass path to toolchain for luajit and curl   | 0000c42bb                                       |
| ea5929db4          | build: fix OpenSSL linking problems on FreeBSD      | caeee019d                                       |
| 7e51aebb9          | build: link libcurl statically from a submodule     | 7b9f44f05                                       |

(There is also 4a22b90866dfda4252577aac4dd7252e518490ae, which is only in 1.10, but it is needed only for autotools libcurl build, so it is irrelevant for master.)

As I see we should do something with:

* zlib -- see the separate comment above,
* nghttp2 -- maybe replace the relevant code with a comment about 1.10/master differences (or just drop it),
* two new master commits -- just copy the new BuildLibCURL.cmake file from the fresh master; those are important fixes.

File: cmake/BuildLibCURL.cmake
Comment (Totktonada) [inline cmake/BuildLibCURL.cmake]: Nit: I would add explicit `-DUSE_NGHTTP2=OFF` (not it is disabled, because it is default).

I mean, replace this block with:

```cmake
# <..a comment re http2, tarantool 1.10 and 2.x..>
list(APPEND LIBCURL_CMAKE_FLAGS "-DUSE_NGHTTP2=OFF")
```

File: cmake/BuildLibCURL.cmake
Comment (Totktonada) [inline cmake/BuildLibCURL.cmake:181]: It seems, this block is dropped by a mistake: it is not related to HTTP/2.0 and libnghttp2.

**Review by Totktonada — APPROVED** [review]
Thanks for the fixes!

LGTM after fixing a few remaining comments.

**Review by igormunkin — APPROVED** [review]
@ligurio, thanks for the patch! Glad, we finally dropped this inconsistency. LGTM.

**Pull request merged**
